### PR TITLE
A *lot* of import fixes for flatpak

### PR DESCRIPTION
**Pull request by powpingdone**
Aside, thank you @hadess for the majority of these fixes. You are the base point for a lot of issues fixed during the creation of this flatpak.

### CODE REVIEW

**Comment by powpingdone** [comment]
Also thank you @stuomas for the starting point for checking BambuStudio.

**Comment by powpingdone** [comment]
(mentioning #1670 because soon(tm))

File: src/slic3r/Utils/PresetUpdater.cpp [hunk 1]
```diff
@@ -818,9 +818,9 @@ void PresetUpdater::priv::sync_tooltip(std::string http_url, std::string languag
         fs::path cache_root = fs::path(data_dir()) / "resources/tooltip";
         try {
             auto vf = cache_root / "common" / "version";
-            if (fs::exists(vf)) load_string_file(vf, common_version);
+            if (fs::exists(vf)) fs::load_string_file(vf, common_version);
```
Comment (SoftFever) [inline src/slic3r/Utils/PresetUpdater.cpp:821]: This will very like to break the build as Boost 1.8.4 has removed this function.
And Orca has migrated to Boost 1.8.4 recently.
So now we have re-implemented this function in src/libslic3r/Utils.hpp:661
We should revert this change

Comment (powpingdone) [reply src/slic3r/Utils/PresetUpdater.cpp:821]: 1.84?

Comment (SoftFever) [reply src/slic3r/Utils/PresetUpdater.cpp:821]: https://www.boost.org/users/history/version_1_84_0.html

Comment (powpingdone) [reply src/slic3r/Utils/PresetUpdater.cpp:821]: That's what I thought.

**Review by SoftFever — APPROVED** [review]
LGTM
Thanks

**Comment by ChaosBlades** [comment]
https://github.com/SoftFever/OrcaSlicer/issues/2747 Flatpak will add Steam Deck compatibility

A Flatpak should also close a lot of various Linux dependency, distro compatibility, performance, and bug issues specific to linux.

It will also show up on a lot of distro software centers once it gets added to flathub. Which should increase OrcaSlicer popularity a bit.

REALLY excited for this. I will finally feel like a real Linux user and not have to go to websites to download updates for any application I use anymore. Will also be able to uninstall AppImage Launcher.

**Comment by tsmith35** [comment]
Just grabbed this build for Windows 10 x64 and got an unhandled unknown exception. Event viewer for the crash was this:

Faulting application name: orca-slicer.exe, version: 1.0.0.0, time stamp: 0x65ba279c
Faulting module name: OrcaSlicer.dll, version: 0.0.0.0, time stamp: 0x65ba2792
Exception code: 0xc000041d
Fault offset: 0x0000000002b735d0
Faulting process id: 0x765c
Faulting application start time: 0x01da569ee641fc48
Faulting application path: C:\Program Files\OrcaSlicer\orca-slicer.exe
Faulting module path: C:\Program Files\OrcaSlicer\OrcaSlicer.dll
Report Id: fadaa8ef-97ac-4500-b8de-c6999b751036
Faulting package full name: 
Faulting package-relative application ID: 

and from the appcrash .wer file:

Sig[0].Name=Application Name
Sig[0].Value=orca-slicer.exe
Sig[1].Name=Application Version
Sig[1].Value=1.0.0.0
Sig[2].Name=Application Timestamp
Sig[2].Value=65ba279c
Sig[3].Name=Fault Module Name
Sig[3].Value=OrcaSlicer.dll
Sig[4].Name=Fault Module Version
Sig[4].Value=0.0.0.0
Sig[5].Name=Fault Module Timestamp
Sig[5].Value=65ba2792
Sig[6].Name=Exception Code
Sig[6].Value=c000041d
Sig[7].Name=Exception Offset
Sig[7].Value=0000000002b735d0

image

**Comment by hadess** [comment]
> Aside, thank you @hadess for the majority of these fixes. You are the base point for a lot of issues fixed during the creation of this flatpak.

Not sure how I wasn't notified on this (I noticed it when a fork of OrcaSlicer updated through a mega squashed patch!?), but glad I could be of help.

It took me a long while to fix all the little bits, so if anyone is feeling generous, I have a [3D printing related wishlist here](https://www.amazon.fr/hz/wishlist/ls/3A4WBDMGAOEDS?type=wishlist&filter=unpurchased&sort=price-asc&viewType=list).
